### ISSUE TITLE
Update CO-OPS SOS python notebook 

### ISSUE THREAD

**Issue by MathewBiddle** [ioos/ioos_code_lab#46]
* xref: https://github.com/ioos/notebooks_demos/issues/404

Update [2016-10-12-fetching_data
](https://ioos.github.io/ioos_code_lab/content/code_gallery/data_access_notebooks/2016-10-12-fetching_data.html) to use non SOS API. 

From @LindsayRAbrams:
From discussions with IT, either our Data API or ERDDAP server would work, depending on what is more useful for your users.
More info on CO-OPS Data API: https://api.tidesandcurrents.noaa.gov/api/prod/
More info on CO-OPS ERDDAP: https://opendap.co-ops.nos.noaa.gov/erddap/index.html

**Comment by ocefpaf** [comment]
I'm trying the ERDDAP server first but it seems that the way the data is organized there we lose some of the best features ERDDAP has. For example, I'm trying to build a URL with these constraints:

```python
from erddapy import ERDDAP

e = ERDDAP(
    server="https://opendap.co-ops.nos.noaa.gov/erddap",
    protocol="tabledap",
)

e.constraints = {
    "time>=": "2016-10-04",
    "time<=": "2016-10-12",
    "longitude>=": -87.40,
    "longitude<=": -74.70,
    "latitude>=": 24.25,
    "latitude<=": 24.25,
}

e.dataset_id = "IOOS_Hourly_Height_Verified_Water_Level"
e.variables = ["WL_VALUE"]
```

I get the following URL (one can construct that using the web interface too):

[https://opendap.co-ops.nos.noaa.gov/erddap/tabledap/IOOS_Hourly_Height_Verified_Water_Level.htmlTable?STATION_ID,latitude,longitude,BEGIN_DATE,END_DATE,time,WL_VALUE&latitude>=24.25&latitude<=24.25&longitude>=-87.40&longitude<=-74.70&BEGIN_DATE="2016-10-04"&END_DATE="2016-10-12"](https://opendap.co-ops.nos.noaa.gov/erddap/tabledap/IOOS_Hourly_Height_Verified_Water_Level.htmlTable?STATION_ID,latitude,longitude,BEGIN_DATE,END_DATE,time,WL_VALUE&latitude>=24.25&latitude<=24.25&longitude>=-87.40&longitude<=-74.70&BEGIN_DATE="2016-10-04"&END_DATE="2016-10-12")

and that fails with:

```
Error {
    code=500;
    message="Internal Server Error: ERROR from data source: com.sybase.jdbc4.jdbc.SybSQLException: Procedure GET_ERDDAP_WL_HOURLY_VFD expects parameter @STATION_ID, which was not supplied.";
}
```


That means I need to know the `STATION_ID`s before hand but that kind of defeats the purpose of an ERDDAP constrained URL :-/

**Comment by MathewBiddle** [comment]
@LindsayRAbrams any ideas why `STATION_ID` is a required constraint on these datasets?

**Comment by ocefpaf** [comment]
BTW, parsing the stations from the recommended URL [1] in the ERDDAP (see the attached image below) fails b/c we cannot access the style sheet. So this fails:

```python
url = "https://opendap.co-ops.nos.noaa.gov/stations/stationsXML.jsp"

pd.read_xml(
    "stationsXML.jsp",
    namespaces={
        "xmlns": "https://stations-prod.co-ops-aws-east1.net/stations/",
        "xsi": "http://www.w3.org/2001/XMLSchema-instance",
    },
    stylesheet="https://stations-prod.co-ops-aws-east1.net/stations/xml_schemas/stations.xsd",
)
```

Maybe there is another source for that info or we can [try to be smart and reverse engineer the style](https://nbviewer.org/gist/ocefpaf/95263c190c0abfc540d8fbede41b82cd). However, all that kind of beats the purpose of having the data in ERDDAP.

[1] https://opendap.co-ops.nos.noaa.gov/stations/stationsXML.jsp
[2] https://stations-prod.co-ops-aws-east1.net/stations/xml_schemas/stations.xsd

![Screenshot from 2022-05-27 09-38-46](https://user-images.githubusercontent.com/950575/170701116-2c67a094-df4a-42b6-a21e-eaa7ea760eeb.png)

**Comment by ocefpaf** [comment]
@LindsayRAbrams the main question here is "how to fetch data from co-ops" if I have to know the station id beforehand, most of the querying tools that we are used to, using only a bounding box and time span, won't work. My guess (and I'm not an ERDDAP expert), is that one would need to re-structure the data on the server to make queries that fit that model. However, after we opened this issue I found out about this project:

https://searvey.readthedocs.io/en/latest/coops.html#co-ops-tidal-station-data

In a way it wraps the current data model in the server and allow for higher level searches like I [mentioned above](coops_stations_within_region). We would not need this library it the data in server is re-arranged but we do have a workaround in searvey in case that is not possible,
